File: 24-Java/src/Progetto.java
import java.util.*;

public class Progetto {
    protected int codice;
    protected String tipo, titolo, nomeCognome, organizzazione;
    protected double milioni;
    private List<ImpegoOrario> impegno;

    
    public Progetto(int codice, String tipo, String titolo, String nomeCognome, String organizzazione, double milioni) {
        this.codice = codice;
        this.tipo = tipo;
        this.titolo = titolo;
        this.nomeCognome = nomeCognome;
        this.organizzazione = organizzazione;
        this.milioni = milioni;
        impegno = new LinkedList<ImpegoOrario>();
    }

    public void addImpegno(ImpegoOrario imp){
        impegno.add(imp);
    }

    public String getTitolo(){
        return titolo;
    }

    public int getCodice() {
        return codice;
    }

    public String toString(){
        return tipo +"\t"+ titolo +"\t"+ codice +"\t"+ nomeCognome +"\t"+ organizzazione;
    }
}

File: 24-Python/Progetto.py
class Progetto:
    def __init__(self, codice, tipo, titolo, nomeCognome, coordinatore, milioni) -> None:
        self._codice = codice
        self._tipo = tipo
        self._titolo = titolo
        self._nomeCognome = nomeCognome
        self._coordinatore = coordinatore
        self._milioni = milioni
        self._impegno = []

    def getCodice(self):
        return self._codice
    
    def addImpegno(self, impegno):
        self._impegno.append(impegno)

    def getTitolo(self):
        return self._titolo

    def __str__(self) -> str:
        return self._tipo +"\t"+ self._titolo +"\t"+ str(self._codice) +"\t"+ self._nomeCognome +"\t"+ self._coordinatore